### PR TITLE
Add support for input and gestures for hand controllers

### PR DESCRIPTION
**Pull request by FejZa**
## Overview

Companion PR to https://github.com/XRTK/XRTK-Core/pull/552

### CODE REVIEW

File: XRTK.Oculus/Packages/com.xrtk.oculus/CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +0,0 @@
-# Changelog
-
-All notable changes to this project will be documented in this file.
```
Comment (StephenHodgson) [inline XRTK.Oculus/Packages/com.xrtk.oculus/CHANGELOG.md:3]: Is this file completely gone or is just the lines removed?

File: XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/InputSystem/Controllers/OculusControllerDataProvider.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@
 using XRTK.Providers.Controllers;
 using XRTK.Services;
 
-namespace XRTK.Oculus.Providers.Controllers
+namespace XRTK.Oculus.Providers.InputSystem.Controllers
```
Comment (StephenHodgson) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/InputSystem/Controllers/OculusControllerDataProvider.cs:16]: Controllers aren't part of the input system. Even in core they're in their own folder:
image

File: XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Profiles.meta [hunk 3]
```diff
@@ -1,5 +1,5 @@
 fileFormatVersion: 2
-guid: 63cd1f630a3d28f4a95eb15df714bd49
+guid: 752641a1dfaf7374db19043f00a59ff3
```
Comment (StephenHodgson) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Profiles.meta:2]: We can probably get rid of this folder now

File: XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Profiles/OculusControllerDataProviderProfile.cs [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 using XRTK.Definitions.Controllers;
 using XRTK.Definitions.Utilities;
-using XRTK.Oculus.Providers.Controllers;
+using XRTK.Oculus.Providers.InputSystem.Controllers;
```
Comment (FejZa) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Profiles/OculusControllerDataProviderProfile.cs:6]: Need to undo this

File: XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/InputSystem/Controllers/BaseOculusController.cs [hunk 5]
```diff
@@ -13,7 +13,7 @@
 using XRTK.Providers.Controllers;
 using XRTK.Services;
 
-namespace XRTK.Oculus.Providers.Controllers
+namespace XRTK.Oculus.Providers.InputSystem.Controllers
```
Comment (FejZa) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/InputSystem/Controllers/BaseOculusController.cs:16]: Undo

File: XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Utilities/OculusHandDataConverter.cs [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+            // The Pinky/Thumb 1+ bones depend on the Pinky/Thumb 0 bone
+            // to be available, which the XRTK hand tracking does not use. We still have to compute them to
+            // be able to resolve pose relation dependencies.
+            if (bone.Id == OculusApi.BoneId.Hand_Thumb1)
+            {
+                GetJointPose(handedness, handSkeleton.Bones[(int)OculusApi.BoneId.Hand_Thumb0]);
+            }
+            else if (bone.Id == OculusApi.BoneId.Hand_Pinky1)
+            {
+                GetJointPose(handedness, handSkeleton.Bones[(int)OculusApi.BoneId.Hand_Pinky0]);
```
Comment (StephenHodgson) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Utilities/OculusHandDataConverter.cs:204]: Do we need to return the data from these recursive calls?

**Review by SimonDarksideJ — APPROVED** [review]

**Pull request merged**